### PR TITLE
Exchange methods

### PR DESCRIPTION
**Pull request by Weistrand**
Declare, delete, bind and unbind exchanges

### CODE REVIEW

File: lib/amqp/client/connection.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ def close(reason = "", code = 200)
       write_bytes FrameBytes.connection_close(code, reason)
       expect(:close_ok)
       @closed = true
+      @socket.close
```
Comment (carlhoerberg) [inline lib/amqp/client/connection.rb:30]: Is this needed? Isn't the socket closed in the `read_loop` after a `close_ok` frame? Or maybe this is the better place to have it in?

File: lib/amqp/client/frames.rb [hunk 2]
```diff
@@ -119,6 +119,87 @@ def channel_close_ok(id)
...
+        1, # type: method
+        id, # channel id
+        frame_size, # frame size
+        40, # class: exchange
+        10, # method: declare
+        0, # reserved1
+        name.bytesize, name,
+        type.bytesize, type,
+        bits,
+        0, # arguments
```
Comment (carlhoerberg) [inline lib/amqp/client/frames.rb:141]: Add `Table` support here, just like in `exchange_bind`

File: lib/amqp/client/frames.rb [hunk 3]
```diff
@@ -119,6 +119,87 @@ def channel_close_ok(id)
...
+        binding_key.bytesize, binding_key,
+        no_wait ? 1 : 0,
+        tbl.bytesize, tbl, # arguments
+        206 # frame end
+      ].pack("C S> L> S> S> S> Ca* Ca* Ca* C L>a* C")
+    end
+
+    def exchange_unbind(id, destination, source, binding_key, no_wait, arguments)
+      tbl = Table.encode(arguments)
+      frame_size = 2 + 2 + 2 + 1 + destination.bytesize + 1 + source.bytesize + 1 + binding_key.bytesize + 1 + 4 + tbl.bytesize
```
Comment (carlhoerberg) [inline lib/amqp/client/frames.rb:186]: can split this line into more lines so it doesn't get so long

**Review by carlhoerberg — COMMENTED** [review]
Well done!

File: lib/amqp/client.rb [hunk 4]
```diff
@@ -25,7 +25,7 @@ def initialize(uri)
 
     def connect
       socket = Socket.tcp @host, @port, connect_timeout: 20, resolv_timeout: 5
-      enable_tcp_keepalive(socket)
+      # enable_tcp_keepalive(socket)
```
Comment (carlhoerberg) [inline lib/amqp/client.rb:28]: let's not have this commented out in the PR

File: lib/amqp/client/connection.rb (hunk 1 above)
Comment (Weistrand) [reply lib/amqp/client/connection.rb:30]: I think we added this together actually, not sure why, I'll remove it

File: lib/amqp/client/frames.rb (hunk 3 above)
Comment (Weistrand) [reply lib/amqp/client/frames.rb]: https://github.com/cloudamqp/amqp-client.rb/pull/2/commits

File: lib/amqp/client/frames.rb (hunk 2 above)
Comment (Weistrand) [reply lib/amqp/client/frames.rb]: https://github.com/cloudamqp/amqp-client.rb/pull/2/commits

**Review by carlhoerberg — APPROVED** [review]

**Pull request merged**